src/Control/Carrier/Fail/Either.hs

```diff
@@ -1,4 +1,9 @@
 {-# LANGUAGE FlexibleInstances, GeneralizedNewtypeDeriving, MultiParamTypeClasses, TypeOperators, UndecidableInstances #-}
+-- | A carrier for a 'Fail' effect, returning the result as an 'Either' 'String'. Failed computations
+-- will return a 'Left' containing the 'String' value passed to 'Control.Monad.Fail.fail'.
+--
+-- Note that 'Fail' effects can, when they are the last effect in a stack, be interpreted directly
+-- to an 'Either' without a call to 'runFail'.
```

Comment:
This is incorrect; we’ve defined the instance s.t. `Either` interprets `Error`, not `Fail`. That may change if #247 replaces `Fail` with `Throw`, but that’s not fully explored yet.